web/packages/teleport/src/UnifiedResources/SearchPanel.tsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import styled from 'styled-components';
+import { Text, Flex } from 'design';
+import { PredicateDoc } from 'shared/components/Search/PredicateDoc';
+
+import Toggle from 'teleport/components/Toggle';
+
+import useServersideSearchPanel, {
+  State,
+  Props,
+} from 'teleport/components/ServersideSearchPanel/useServerSideSearchPanel';
+import Tooltip from 'teleport/components/ServersideSearchPanel/Tooltip';
+
+import SearchInput from './SearchInput';
+
+export default function Container(props: Props) {
+  const state = useServersideSearchPanel(props);
+  return <SearchPanel {...state} />;
+}
+
+// Adapted from teleport.components.ServersideSearchPanel
+export function SearchPanel({
+  searchString,
+  setSearchString,
+  isAdvancedSearch,
+  setIsAdvancedSearch,
+  onSubmitSearch,
+}: State) {
+  function onToggle() {
+    setIsAdvancedSearch(!isAdvancedSearch);
```

Comment:
It is safer to use an [updater function](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state):
```suggestion
    setIsAdvancedSearch(wasAdvancedSearch => !wasAdvancedSearch);
```